### PR TITLE
feat: Add checktable widget

### PR DESCRIPTION
**Pull request by Kondrahin**

### CODE REVIEW

File: pybotx_widgets/checktable.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    #: command that will be triggered by click on the element.
+    command: str
+    #: value which will be displayed on the button associated with this checkbox.
+    checkbox_value: Any = None
+    #: list of available values.
+    mapping: Optional[Dict[Any, str]] = None
+    #: extra payload that will be stored in button and then received in new message.
+    data: Optional[Dict[str, Any]] = None
+    #: can the value be Null
+    nullable: bool = False
```
Comment (vonabarak) [inline pybotx_widgets/checktable.py:25]: Suggest adding some validation to ensure that checkbox_value is not None when nullable=False.
Mb pydantic model is more suitable from this point of view.

**Review by vonabarak — APPROVED** [review]

File: pybotx_widgets/checktable.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    label: str
+    #: command that will be triggered by click on the element.
+    command: str
+    #: value which will be displayed on the button associated with this checkbox.
+    checkbox_value: Union[T, Undefined] = undefined
+    #: list of available values.
+    mapping: Optional[Dict[T, str]] = None
+    #: extra payload that will be stored in button and then received in new message.
+    data: Optional[Dict[str, Any]] = None
+    #: can the value be Null
```
Comment (lxmnk) [inline pybotx_widgets/checktable.py:26]: `Null` -> `None`.

File: pybotx_widgets/checktable.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    nullable: bool = False
```
Comment (lxmnk) [inline pybotx_widgets/checktable.py:27]: Don't you forget to drop this property?

Comment (Kondrahin) [reply pybotx_widgets/checktable.py:27]: My fault(

**Review by lxmnk — APPROVED** [review]

**Review by Nimond — APPROVED** [review]

File: pybotx_widgets/checktable.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+    @root_validator(pre=True)
+    def is_checkbox_value_exist_in_mapping(cls, values):
+        mapping = values.get("mapping")
+        checkbox_value = values.get("checkbox_value")
+        if (
+                mapping
+                and not isinstance(checkbox_value, Undefined)
+                and checkbox_value not in mapping
+        ):
```
Comment (d-tamirlan) [inline pybotx_widgets/checktable.py:38]: nit: mb replace with this:
```python
error_conditions = (
    mapping,
    not isinstance(checkbox_value, Undefined),
    checkbox_value not in mapping
)
if all(error_conditions):
    pass

File: pybotx_widgets/undefined.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+"""Singleton for `undefined` value."""
+
+
+class Undefined:
+    def __new__(cls) -> "Undefined":
```
Comment (d-tamirlan) [inline pybotx_widgets/undefined.py:5]: You can just use `lru_cache`
```python
class Undefined:
    @lru_cache()
    def __new__(cls) -> "Undefined":
        super().__new__(cls)
```

File: pybotx_widgets/checktable.py (hunk 4 above)
Comment (Kondrahin) [reply pybotx_widgets/checktable.py:38]: In this implementation, if `mapping == None`, an error occurs, since mapping not iterable object.

**Review by d-tamirlan — APPROVED** [review]

**Pull request merged**